src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl

```diff
@@ -916,57 +956,72 @@ inline void FUNC(fc_bf_tiled_kernel_dyn_quan)(
+                    dq_wei_unpacked_tmp.s4567 = (UNPACK_INT4(DQ_TYPE, *((INT4_PACKED_TYPE_PRELOAD*)&wei_packed1)));
+                    dq_wei_unpacked.s01 = dq_wei_unpacked_tmp.s01;
+                    dq_wei_unpacked.s23 = dq_wei_unpacked_tmp.s45;
+                    dq_wei_unpacked.s45 = dq_wei_unpacked_tmp.s23;
+                    dq_wei_unpacked.s67 = dq_wei_unpacked_tmp.s67;
+                #else
+                    SLM_FILTER_PACKED_VEC wei_packed = BLOCK_READN(FILTER_TYPE, FILTER_LOAD_BLOCK_SIZE, weights, weights_idx);
+                    SLM_WEIGHT_UNPACKED_VEC dq_wei_unpacked = (UNPACK_TRANSPOSED_INT4(DQ_TYPE, *((INT4_PACKED_TYPE_PRELOAD *)&wei_packed)));
+                #endif
+            #else  // COMPRESSED_WEIGHTS_INT8
+                    SLM_WEIGHT_UNPACKED_VEC dq_wei_unpacked;
+                    WEIGHT_VEC_TYPE wei_packed = TO_WEIGHT_VEC_TYPE(FILTER_BLOCK_READ(weights, weights_idx));
+                    dq_wei_unpacked.s0123 = wei_packed.s0246;
+                    dq_wei_unpacked.s4567 = wei_packed.s1357;
             #endif
 
-            // Calculate zero-point and scale only for DECOMPRESSION_SCALE_POST_OP enabled
+            // Calculate zero-point and scale only for DQ_DECOMPRESSION_SCALE_POST_OP enabled
             // Calculate weight : w = (w - dzp) * ds
             // if DECOMPRESSION_ZP_TERM is not enabled, then dzp is ACCUMULATOR_VAL_ZERO.
-            #if DECOMPRESSION_ZP_TERM
+            #if DECOMPRESSION_ZP_TERM && !COMPRESSED_WEIGHTS_INT8
                 #if DECOMPRESSION_ZP_SCALAR
-                    DQ_SLM_FILTER_UNPACKED_VEC dzp = (DQ_SLM_FILTER_UNPACKED_VEC)(DECOMPRESSION_ZP_VALUE);
+                    SLM_WEIGHT_UNPACKED_VEC dzp = (SLM_WEIGHT_UNPACKED_VEC)(DECOMPRESSION_ZP_VALUE);
                     dq_wei_unpacked -= dzp;
                 #elif DECOMPRESSION_ZP_GROUPS_NUM > 1
-                    DQ_TYPE* w = (DQ_TYPE*)(&dq_wei_unpacked);
+                    FILTER_TYPE* w = (FILTER_TYPE*)(&dq_wei_unpacked);
                     const uint ni_offset = ni * TILE_IFM * SIMD + local_id * FILTER_LOAD_ITERS * FILTER_LOAD_BLOCK_SIZE;
                     unroll_for(uint fi = 0; fi < TILE_OFM; ++fi) {
                         const uint offset_ofm = out_f + fi*SIMD + sglid;
                         unroll_for(uint kii = 0; kii < FILTER_LOAD_BLOCK_SIZE; ++kii) {
                             const uint offset_ifm = ni_offset + load_iter * FILTER_LOAD_BLOCK_SIZE + kii;
                             const uint zp_offset = (offset_ofm % DECOMPRESSION_ZP_BATCH_NUM) * DECOMPRESSION_ZP_BATCH_PITCH +
                                                     (offset_ifm / DECOMPRESSION_ZP_GROUP_SIZE) * DECOMPRESSION_ZP_FEATURE_PITCH;
-                            w[W_DYN_QUAN_IDX] = w[W_DYN_QUAN_IDX] - TO_DQ_TYPE(decompression_zp[zp_offset]);
+                            w[W_DYN_QUAN_IDX] = w[W_DYN_QUAN_IDX] - CAT(CAT(convert_, FILTER_TYPE),_rte)(decompression_zp[zp_offset]);
```

Comment:
nit: I think now we can remove type conversion here because decompression_zp is also integral type..

Reply:
Applied